pkg/deploy/vmsscleaner/clean.go

```diff
@@ -0,0 +1,50 @@
+package vmsscleaner
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/compute"
+)
+
+type Interface interface {
+	RemoveFailedNewScaleset(ctx context.Context, rgName, vmssName string) (deleted bool)
+}
+
+type cleaner struct {
+	log  *logrus.Entry
+	vmss compute.VirtualMachineScaleSetsClient
+}
+
+func New(log *logrus.Entry, vmss compute.VirtualMachineScaleSetsClient) Interface {
+	return &cleaner{
+		log:  log,
+		vmss: vmss,
+	}
+}
+
+func (c *cleaner) RemoveFailedNewScaleset(ctx context.Context, rgName, vmssName string) (deleted bool) {
```

Comment:
I like how the function is clean to read. I would have a small suggestion to make it more streamlined

```suggestion
func (c *cleaner) RemoveScaleset(ctx context.Context, rgName, vmssName string) (deleted bool) {
```

as the function deletes everything passed in.

Reply:
>the function deletes everything passed in
Not true. It only deletes `vmssName`, and it only does this if there are at least two VMSSes. Otherwise, it does nothing.